### PR TITLE
Render layer selector from config

### PR DESCRIPTION
**Pull request by kdeloach**
This PR handles schema validation and (basic) rendering the tree to the UI. I will create a separate PR for styling and user interactions.

image

Connects #479

### CODE REVIEW

**Comment by mmcfarland** [comment]
Looks great and simple.  My recollection was that we switched _from_ `name` to `layerName`/`serviceName`, not the other way around and the high level `layers` was different from the `include/excludeLayers`.  This looks like it works though, and makes sense.  I'll defer to @caseypt if he sees an issue with it.

File: src/GeositeFramework/plugins/layer_selector_v2/config.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            location: "//cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3",
+            main: "underscore-min"
+        }
+    ]
+});
+
+define([
+        "dojo/_base/declare",
+        "dojo/json",
+        "use!tv4",
```
Comment (mmcfarland) [inline src/GeositeFramework/plugins/layer_selector_v2/config.js]: We were looking at dropping `use` in favor of the dojo shim system that's now available.  Consider upgrading this.

**Comment by kdeloach** [comment]
> Looks great and simple. My recollection was that we switched from name to layerName/serviceName, not the other way around and the high level layers was different from the include/excludeLayers. This looks like it works though, and makes sense. I'll defer to @caseypt if he sees an issue with it.

I suggested using layerName/serviceName but as I was implementing this I realized that the original names were better. The longer names are a little redundant and it was confusing how "server" and "service" is used interchangeably.

**Comment by caseypt** [comment]
This looks great. Config looks good to me.

File: src/GeositeFramework/plugins/layer_selector_v2/config.js (hunk 1 above)
Comment (kdeloach) [inline src/GeositeFramework/plugins/layer_selector_v2/config.js]: I'd like to do this in a separate PR when I have a stronger grasp on what this entails.

Comment (caseypt) [inline src/GeositeFramework/plugins/layer_selector_v2/config.js]: It looks like it's AMD compliant now: https://github.com/geraintluff/tv4#installation

**Comment by mmcfarland** [comment]
Cool, +1

File: src/GeositeFramework/plugins/layer_selector_v2/config.js (hunk 1 above)
Comment (kdeloach) [inline src/GeositeFramework/plugins/layer_selector_v2/config.js]: I tried to load the latest release of tv4 as a package (similar to jquery or underscore) but it didn't work as I expected it to.
